dist/schema-form.js

```diff
@@ -226,6 +231,67 @@ angular.module('schemaForm').provider('schemaFormDecorators',
               }
             };
 
+            var evalExpression = function (expression) {
+              angular.forEach(scope.form.dependencies, function (key) {
+                expression = expression.replace(key, lookupForKey(key));
+              });
+
+              return scope.$eval(expression);
+            };
+
+
+            var resetModelValue = function () {
+              if (scope.ngModel) {
+                scope.ngModel.$setViewValue(undefined);
+                scope.ngModel.$render();
+                scope.ngModel.$setPristine();
+              }
+            };
+
+            scope.showCondition = function () {
+              var expressionString = scope.form.expression;
+              if (angular.isUndefined(expressionString)) {
+                return true;
+              }
+
+              var show = evalExpression(expressionString);
+
+              if (angular.isDefined(scope.form.required)) {
+                scope.form.required = show;
+                scope.form.schema.required = show;
+              }
+
+
+              if (scope.form.key && !show) {
+                resetModelValue();
+              }
+
+              return show;
+
+            };
+
+
+            scope.disabledElement = function () {
+              var expressionString = scope.form.disabledExpression;
+
+              if (angular.isUndefined(expressionString)) {
+                return false;
+              }
+
+              var disabled = evalExpression(expressionString);
+
+              if (angular.isDefined(scope.form.required)) {
+                scope.form.required = !disabled;
+                scope.form.schema.required = !disabled;
```

Comment:
This is kind of a restriction we have by using JSON Schema. It becomes a bit complicated when your validation scenario is hard to describe with a JSON Schema.

What you can do is to re-redraw the form by changing the schema or the form definition, but change the entire schema/form instance (i.e. use angular.copy), otherwise it won't trigger a redraw of the form. 

A bit more correct solution would be to use `oneOf`, `anyOf` etc and that way you might be able to describe the validation you need from within the schema, but sadly we lack support for those. 

Could you give some more specifics on what such a situation might be? There might be some other way to solve it.